### PR TITLE
Use stubs instead of wrappers in generated test headers when needed

### PR DESCRIPTION
**Pull request by IDKWNTCMF**

### CODE REVIEW

File: server/src/clang-utils/SourceToHeaderRewriter.cpp [hunk 1]
```diff
@@ -133,19 +136,28 @@ std::string SourceToHeaderRewriter::generateWrapper(const fs::path &sourceFilePa
...
+    ExecUtils::doWorkWithProgress(tests, progressWriter, logMessage,
+                                  [this, &stubGen, &targets](auto &it) {
         fs::path const &sourceFilePath = it.first;
-        tests::Tests &test = const_cast<tests::Tests &>(it.second);
-        test.headerCode = generateTestHeader(sourceFilePath, test);
+        auto &test = const_cast<tests::Tests &>(it.second);
+        auto iterator = targets.find(sourceFilePath);
+        bool externFromStub =
+            iterator != targets.end() &&
+            CollectionUtils::contains(stubGen.getStubSources(iterator->second),sourceFilePath);
```
Comment (ladisgin) [inline server/src/clang-utils/SourceToHeaderRewriter.cpp:160]: not formatted

File: server/src/Server.cpp [hunk 2]
```diff
@@ -265,6 +262,10 @@ Status Server::TestsGenServiceImpl::ProcessBaseTestRequest(BaseTestGen &testGen,
         Linker linker{testGen, stubGen, lineInfo, generator};
         linker.prepareArtifacts();
         auto testMethods = linker.getTestMethods();
+        auto successfulTargets = linker.getSuccessfulTargets();
```
Comment (ladisgin) [inline server/src/Server.cpp:265]: bad naming

File: server/src/clang-utils/SourceToHeaderRewriter.cpp [hunk 3]
```diff
@@ -133,19 +136,28 @@ std::string SourceToHeaderRewriter::generateWrapper(const fs::path &sourceFilePa
...
+                                                 ProgressWriter const *progressWriter) {
     std::string logMessage = "Generating headers for tests";
     LOG_S(DEBUG) << logMessage;
-    ExecUtils::doWorkWithProgress(tests, progressWriter, logMessage, [this](auto &it) {
+    ExecUtils::doWorkWithProgress(tests, progressWriter, logMessage,
+                                  [this, &stubGen, &targets](auto &it) {
         fs::path const &sourceFilePath = it.first;
-        tests::Tests &test = const_cast<tests::Tests &>(it.second);
-        test.headerCode = generateTestHeader(sourceFilePath, test);
+        auto &test = const_cast<tests::Tests &>(it.second);
```
[the remark this answers is not in the document]
Comment (IDKWNTCMF) [reply server/src/clang-utils/SourceToHeaderRewriter.cpp:156]: Here `it` has `const std::pair<fs::path, Tests> &` type as a lambda parameter, but we want to modify `it.second` to save generated header code.

File: server/src/clang-utils/SourceToHeaderRewriter.cpp (hunk 1 above)
Comment (IDKWNTCMF) [reply server/src/clang-utils/SourceToHeaderRewriter.cpp]: Refactored

File: server/src/Server.cpp (hunk 2 above)
Comment (IDKWNTCMF) [reply server/src/Server.cpp]: Renamed to `selectedTargets`

**Review by ladisgin — APPROVED** [review]

**Pull request merged**